core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -1005,23 +1056,171 @@ private void processBind(@Transient ObjList<BindVariableSetter> bindVariableSett
+            lo = lo + 1;
+            long hi = getStringLength(lo, msgLimit);
+            checkNotTrue(hi == -1, "bad prepared statement name length");
+            CharSequence statementName = getStatementName(lo, hi);
+            if (statementName != null) {
+                final NamedStatementWrapper wrapper = namedStatementMap.get(statementName);
+                wrapper.clear();
+                bindVarTypesStack.push(wrapper.bindVariableTypes);
+                namedStatementWrapperStack.push(wrapper);
+            }
+        }
 
-    private void processClose() throws PeerDisconnectedException, PeerIsSlowToReadException {
-        prepareForNewQuery();
         sink().put('3'); // close complete
         sink().putNetworkInt(Integer.BYTES);
-        send();
     }
 
-    private void processDescribe() throws PeerDisconnectedException, PeerIsSlowToReadException {
+    private void processDescribe(@Transient ObjList<BindVariableSetter> bindVariableSetters,
+                                 long lo,
+                                 long msgLimit,
+                                 @Transient CharSequenceObjHashMap<NamedStatementWrapper> namedStatementMap) throws SqlException, BadProtocolException {
+        lo = lo + 1;
+        long hi = getStringLength(lo, msgLimit);
+        checkNotTrue(hi == -1, "bad portal name length [msgType='D']");
+
+        CharSequence statementName = getStatementName(lo, hi);
+        setupNamedStatement(bindVariableSetters, namedStatementMap, statementName);
+
         if (currentFactory != null) {
             prepareRowDescription();
-            send();
             LOG.info().$("described").$();
         }
     }
 
+    private void processParse(
+            long address,
+            long lo,
+            long msgLimit,
+            @Transient SqlCompiler compiler,
+            @Transient AssociativeCache<Object> factoryCache,
+            @Transient CharSequenceObjHashMap<NamedStatementWrapper> namedStatementMap,
+            @Transient ObjList<BindVariableSetter> bindVariableSetters
+    ) throws BadProtocolException, SqlException, PeerDisconnectedException, PeerIsSlowToReadException {
+        // 'Parse'
+        // this appears to be the execution side - we must at least return 'RowDescription'
+        // possibly more, check QueryExecutionImpl.processResults() in PG driver for more info
+
+        long hi = getStringLength(lo, msgLimit);
+        checkNotTrue(hi == -1, "bad prepared statement name length");
+
+        CharSequence statementName = getStatementName(lo, hi);
+
+        lo = hi + 1;
+
+        hi = getStringLength(lo, msgLimit);
+        checkNotTrue(hi == -1, "bad query text length");
+
+        prepareForNewQuery();
+        parseQueryText(lo, hi);
+
+        lo = hi + 1;
+
+        checkNotTrue(lo + Short.BYTES > msgLimit, "could not read parameter count");
+
+        short parameterCount = getShort(lo);
+
+        IntList bindVariableTypes = bindVarTypesStack.pop();
```

Comment:
what would happen to this instance if parameterCount == 0 && statementName == null ?